### PR TITLE
Repo url pickers autocomplete filter

### PR DESCRIPTION
**Pull request by cell-kibet**
#### :heavy_check_mark: Checklist

- [ ] Subsituted "Select" in the "pickers" with "Autocomplete" from material-ui.
- [ ] This facilitates filtering of "owners" in case you have a pretty long list

image

### CODE REVIEW

**Comment by freben** [comment]
Hi!

Note that you will need to create changesets for modified packages. See the automated comment above ☝️

File: plugins/scaffolder/src/components/fields/RepoUrlPicker/AzureRepoPicker.tsx [hunk 1]
```diff
@@ -55,15 +57,29 @@ export const AzureRepoPicker = (props: {
...
-            disabled={allowedOrganizations.length === 1}
-            selected={organization}
-            items={organizationItems}
+          <Autocomplete
+          aria-label="Organization"
+          options={organizationItems}
+          getOptionLabel={(organizations) => organizations.label || "error getting label"}
+          disabled={allowedOrganizations.length === 1}
+          data-testid="select"
+          freeSolo
```
Comment (benjdlambert) [inline plugins/scaffolder/src/components/fields/RepoUrlPicker/AzureRepoPicker.tsx:66]: Not sure what we want to have `freeSolo` on here if there's `allowedOrgs`

File: plugins/scaffolder/src/components/fields/RepoUrlPicker/AzureRepoPicker.tsx [hunk 2]
```diff
@@ -55,15 +57,29 @@ export const AzureRepoPicker = (props: {
...
+          onChange={selected =>
+            onChange({
+              organization: String(Array.isArray(selected) ? selected[0] : selected),
+            })
+          } 
+          tabIndex={0}
```
Comment (benjdlambert) [inline plugins/scaffolder/src/components/fields/RepoUrlPicker/AzureRepoPicker.tsx:72]: Probably want to remove this from all the `Autocompletes` to as it affects accessibility?

File: plugins/scaffolder/src/components/fields/RepoUrlPicker/GithubRepoPicker.tsx [hunk 3]
```diff
@@ -42,15 +44,31 @@ export const GithubRepoPicker = (props: {
...
+          onChange={selected =>
+            onChange({
+              owner: String(Array.isArray(selected) ? selected[0] : selected),
+            })
+          } 
+          tabIndex={0}
+          renderInput={params => (
+              <TextField
+              {...params}
+              placeholder="Owner Available"
```
Comment (benjdlambert) [inline plugins/scaffolder/src/components/fields/RepoUrlPicker/GithubRepoPicker.tsx:63]: Think that this should just say the relevant thing you're filtering. Like `Select an Owner`

File: plugins/scaffolder/src/components/fields/RepoUrlPicker/AzureRepoPicker.tsx (hunk 1 above)
Comment (benjdlambert) [reply plugins/scaffolder/src/components/fields/RepoUrlPicker/AzureRepoPicker.tsx:66]: Same for all the other autocompletes

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Some higher level comments here, and also wondering if its possible to still render the dropdown accent like this at all times too (also for accessibility).

image

I think there's some docs that can help you on the [mui docs](https://v4.mui.com/components/autocomplete/#autocomplete)

Screenshot 2023-01-25 at 11 58 28

**Comment by benjdlambert** [comment]
Maybe it would also make sense to be able to default to the first one like it does today?

File: plugins/scaffolder/src/components/fields/RepoUrlPicker/GithubRepoPicker.tsx (hunk 3 above)
Comment (cell-kibet) [reply plugins/scaffolder/src/components/fields/RepoUrlPicker/GithubRepoPicker.tsx:63]: Thank you @benjdlambert . Let me make the changes. Thanks for the support thus far.. :)

File: plugins/scaffolder/src/components/fields/RepoUrlPicker/AzureRepoPicker.tsx (hunk 1 above)
Comment (cell-kibet) [reply plugins/scaffolder/src/components/fields/RepoUrlPicker/AzureRepoPicker.tsx]: Then I will need a bit of help with state management @benjdlambert if you do not mind. Am a bit stuck with onChange

**Comment by cell-kibet** [comment]
> 



> Maybe it would also make sense to be able to default to the first one like it does today?

This is done..

**Comment by freben** [comment]
Note that the tests are failing now.

Also when Vale is complaining about spelling of identifiers, put them in backticks, like so ``` `allowedOwners` ```

**Comment by cell-kibet** [comment]
> 

Alright @freben

**Comment by cell-kibet** [comment]
> Note that the tests are failing now.
> 
> Also when Vale is complaining about spelling of identifiers, put them in backticks, like so `` `allowedOwners` ``

I will need your help on the tests @freben . Am pretty much stuck

**Comment by cell-kibet** [comment]
> Maybe it would also make sense to be able to default to the first one like it does today?

Hey @benjdlambert .. Kindly help me with the tests here.. Everything else works fine now..

**Comment by benjdlambert** [comment]
Looks like the tests are failing as they're looking for text in the document that are not present. You might need to adjust the test code to potentially click some of the dropdown or to check some other elements instead now as the markup has changed.

**Comment by cell-kibet** [comment]
> 

Thanks Ben

**Pull request closed without merging**